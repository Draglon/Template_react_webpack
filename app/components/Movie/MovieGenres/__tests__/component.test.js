import React from 'react';

import MovieGenresComponent from '../component';

describe('<MovieGenresComponent />', () => {
  const props = {
    title: 'some title',
    genres: [],
  };
  const component = shallow(<MovieGenresComponent {...props} />);

  describe('should match snapshots', () => {
    it('should match snapshot - without genres', () => {
      expect(component).toMatchSnapshot();
    });

    it('should match snapshot - with genres', () => {
      const newProps = {
        ...props,
        genres: [{ id: 1, name: 'name_1' }, { id: 2, name: 'name_2' }],
      };
      component.setProps(newProps);
      expect(component).toMatchSnapshot();
    });
  });
});
