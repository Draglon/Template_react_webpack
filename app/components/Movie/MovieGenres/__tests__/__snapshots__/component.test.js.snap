// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieGenresComponent /> should match snapshots should match snapshot - with genres 1`] = `
<Row
  gutter={0}
>
  <Col
    offset={2}
    span={20}
  >
    <Paragraph
      className="movie__text"
    >
      <b>
        some title
        : 
      </b>
      <Tag
        closable={false}
        key="1"
      >
        name_1
      </Tag>
      <Tag
        closable={false}
        key="2"
      >
        name_2
      </Tag>
    </Paragraph>
  </Col>
</Row>
`;

exports[`<MovieGenresComponent /> should match snapshots should match snapshot - without genres 1`] = `
<Row
  gutter={0}
>
  <Col
    offset={2}
    span={20}
  >
    <Paragraph
      className="movie__text"
    >
      <b>
        some title
        : 
      </b>
    </Paragraph>
  </Col>
</Row>
`;
